src/main/java/view/createPostView.java

```diff
@@ -0,0 +1,209 @@
+package view;
+
+import controller.create_post.CreatePostController;
+import controller.create_post.CreatePostState;
+import controller.create_post.CreatePostViewModel;
+import controller.homepage.HomepageController;
+import entity.Comment;
+import entity.User;
+
+import javax.swing.*;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+public class createPostView extends JDialog implements PropertyChangeListener {
+
+    private final String viewName = "create post";
+
+    private JTextField titleField;
+    private JTextArea contentField;
+    private JComboBox<String> categoryDropdown;
+    private JButton saveButton;
+    private JButton cancelButton;
+
+    private final CreatePostViewModel createPostViewModel;
+    private final CreatePostController createPostController;
+    private final HomepageController homepageController;
+
+
+    public createPostView(CreatePostViewModel createPostViewModel, CreatePostController createPostController, HomepageController homepageController) {
+        this.createPostViewModel = createPostViewModel;
+        this.createPostController = createPostController;
+        this.homepageController = homepageController;
+
+        createPostViewModel.addPropertyChangeListener(this);
+
+        setSize(400, 300);
+        setLayout(new BorderLayout());
+        setBackground(StyleConstants.BACKGROUND_COLOR);
+
+        final JLabel title = new JLabel("Creating Post");
+        title.setAlignmentX(Component.CENTER_ALIGNMENT);
+
+        // Input Fields
+        JPanel inputPanel = new JPanel();
+        inputPanel.setLayout(new GridLayout(2, 2));
+
+        inputPanel.add(new JLabel("Title: "));
+        titleField = new JTextField();
+        inputPanel.add(titleField);
+
+        inputPanel.add(new JLabel("Category: "));
+        categoryDropdown = new JComboBox<>(new String[]{"Education", "Politics", "Sports", "Random"});
```

Comment:
Could you add "Java" here too? I have a "Java" filtering view on my slides

Reply:
yes will do!